addons/mail/controllers/discuss.py

```diff
@@ -109,6 +109,8 @@ def _response_discuss_channel_invitation(self, channel_sudo, is_channel_token_se
 
     def _response_discuss_public_channel_template(self, channel_sudo, discuss_public_view_data=None):
         discuss_public_view_data = discuss_public_view_data or {}
+        if channel_sudo.public == 'groups' and channel_sudo._context.get('guest'):
+            raise NotFound()
```

Comment:
Hello, thank you for the PR.

The check at this step is too late, the method actually adding members is `_response_discuss_channel_invitation`. It should probably raise only from inside `add_members`.

Because other calls to `add_members` should not allow guest into such channels either.